core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -42,6 +114,88 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
         }
     );
 
+    readonly setPages = this.updater<DotCMSContentlet[]>(
+        (state: DotPagesState, pages: DotCMSContentlet[]) => {
+            return {
+                ...state,
+                pages: {
+                    ...state.pages,
+                    items: [...pages]
+                }
+            };
+        }
+    );
+
+    readonly setKeyword = this.updater<string>((state: DotPagesState, keyword: string) => {
+        return {
+            ...state,
+            pages: {
+                ...state.pages,
+                items: [],
+                keyword
+            }
+        };
+    });
+
+    readonly setLanguageId = this.updater<string>((state: DotPagesState, languageId: string) => {
+        return {
+            ...state,
+            pages: {
+                ...state.pages,
+                items: [],
+                languageId
+            }
+        };
+    });
+
+    readonly setArchived = this.updater<string>((state: DotPagesState, archived: string) => {
+        return {
+            ...state,
+            pages: {
+                ...state.pages,
+                items: [],
+                archived: !!archived
+            }
+        };
+    });
+
+    readonly clearMenuActions = this.updater((state: DotPagesState) => {
+        return {
+            ...state,
+            pages: {
+                items: [...state.pages.items],
+                actionMenuDomId: null,
+                menuActions: []
+            }
+        };
+    });
+
+    readonly setMenuActions = this.updater(
+        (
+            state: DotPagesState,
+            { actions, actionMenuDomId }: { actions: MenuItem[]; actionMenuDomId: string }
+        ) => {
+            return {
+                ...state,
+                pages: {
+                    ...state.pages,
+                    menuActions: actions,
+                    actionMenuDomId
+                }
```

Comment:
This is not creating a new object inside `items`.

Reply:
on the updater for `setMenuActions` the "items" value is kept the same, only `menuActions` and `actionMenuDomId` are changed